sdk/tables/data-tables/src/TableTransaction.ts

```diff
@@ -266,7 +271,15 @@ export class InternalTableTransaction {
       this.resetableState.transactionId,
       this.resetableState.changesetId
     );
-    const client = new ServiceClient();
+
+    const options: ServiceClientOptions = {
```

Comment:
NIT: I personally found this initialization to be a little confusing, with the weird use of the spread operation.  Could we do:

```typescript
const options: ServiceClientOptions = { };
if (isTokenCredential(this.credential)) {
  options.credentialScopes = STORAGE_SCOPE;
  options.credential: this.credential
}
```

Instead?  I know we use this spread pattern in a few other places but it felt a little better there (as we are constructing a larger object inline.

You could also consider some sort of `serviceClientOptionsForCredential` helper that would return this object for `TokenCredential` and the empty object for other credentials and then just spread the result of calling that (here and in the other places), so people didn't have to work out exactly what this pattern was doing.

Just a thought, this sort of thing may be a common idiom I've just never seen in which case feel free to push back!

Reply:
Thanks Matt, I agree. It reads better the way you suggested, I'm updating this there is no need for the spread.

Sometimes that spread is useful, in scenarios where you are building up an object and one of the properties may be undefined and you don't want to have  `property: undefined`, instead you prefer not to have the property at all.

```typescript
const foo = {
   p1: "one",
   p2: 2,
   ...(mayBeUndefined && {p3: mayBeUndefined})
}
console.log(foo)
// { p1: "one", p2: 2}

const bar= {
   p1: "one",
   p2: 2,
   p3: mayBeUndefined
}
console.log(bar)
// { p1: "one", p2: 2, p3: undefined}
```